subsys/net/ip/net_pkt.c

```diff
@@ -1797,6 +1797,16 @@ int net_pkt_remove(struct net_pkt *pkt, size_t length)
 	return 0;
 }
 
+int net_pkt_pull(struct net_pkt *pkt, size_t length)
+{
+	if (pkt->buffer) {
+		net_buf_pull(pkt->buffer, length);
```

Comment:
We should check here that if the net_buf->len is smaller than what we try to pull, the first net_buf should be removed in that case (and the second one if the pull length is large enough etc)

Reply:
indeed, without this it's a memory leak issue ahead of us.